### PR TITLE
Add LoadMap

### PR DESCRIPTION
**Pull request by hebime**
This adds functionality to load env vars and decrypt them from an in memory map.
It removes trimming space from values.

### CODE REVIEW

**Comment by amineck** [comment]
I'm a bit confused, can you please explain why we need `ParseMap` to also map `Plain` values as well? to keep the same API?
Cause If we are not passing .env files, the ParseMap is only useful pull secrets, right?

File: internal/envparser/envparser.go [hunk 1]
```diff
@@ -113,7 +113,7 @@ func (p *lineParser) parse(envVars *EnvVars, l string) error {
 
 	// key is first index, value is second
 	k := strings.TrimSpace(splits[0])
-	v := strings.TrimSpace(splits[1])
```
Comment (amineck) [inline internal/envparser/envparser.go:116]: Is this going to affect multi line strings?

**Review by amineck — APPROVED** [review]
LGTM

**Comment by hebime** [comment]
> I'm a bit confused, can you please explain why we need `ParseMap` to also map `Plain` values as well? to keep the API stable?
> Cause If we are not passing .env files, the ParseMap is only useful pull secrets, right?

It is slightly redundant in the sense that the original map will have plain text values in it as well. It keeps the API consistent in the sense that it also loads the plain text values into the environment.

File: internal/envparser/envparser.go (hunk 1 above)
Comment (iToto) [reply internal/envparser/envparser.go:116]: Wouldn't this test fail if it did? https://github.com/wingocard/serum/blob/master/internal/envparser/envparser_test.go#L128

**Review by iToto — APPROVED** [review]
💌

**Comment by hebime** [comment]
Closing this in favour of #14

**Pull request closed without merging**